secretstores/akeyless/akeyless.go

```diff
@@ -0,0 +1,539 @@
+package akeyless
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+
+	aws "github.com/akeylesslabs/akeyless-go-cloud-id/cloudprovider/aws"
+	"github.com/akeylesslabs/akeyless-go/v5"
+
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/components-contrib/secretstores"
+	"github.com/dapr/kit/logger"
+	kitmd "github.com/dapr/kit/metadata"
+)
+
+var _ secretstores.SecretStore = (*akeylessSecretStore)(nil)
+
+// akeylessSecretStore is a secret store implementation for Akeyless.
+type akeylessSecretStore struct {
+	v2     *akeyless.V2ApiService
+	token  string
+	logger logger.Logger
+}
+
+// NewAkeylessSecretStore returns a new Akeyless secret store.
+func NewAkeylessSecretStore(logger logger.Logger) secretstores.SecretStore {
+	return &akeylessSecretStore{
+		logger: logger,
+	}
+}
+
+// akeylessMetadata contains the metadata for the Akeyless secret store.
+type akeylessMetadata struct {
+	GatewayURL             string `json:"gatewayUrl" mapstructure:"gatewayUrl"`
+	JWT                    string `json:"jwt" mapstructure:"jwt"`
+	AccessID               string `json:"accessId" mapstructure:"accessId"`
+	AccessKey              string `json:"accessKey" mapstructure:"accessKey"`
+	AccessType             string `json:"accessType" mapstructure:"accessType"`
+	K8SGatewayURL          string `json:"k8sGatewayUrl" mapstructure:"k8sGatewayUrl"`
+	K8SAuthConfigName      string `json:"k8sAuthConfigName" mapstructure:"k8sAuthConfigName"`
+	K8sServiceAccountToken string `json:"k8sServiceAccountToken" mapstructure:"k8sServiceAccountToken"`
+}
+
+// Init creates a new Akeyless secret store client and sets up the Akeyless API client
+// with authentication method based on the accessId.
+func (a *akeylessSecretStore) Init(ctx context.Context, meta secretstores.Metadata) error {
+	a.logger.Info("Initializing Akeyless secret store...")
+	m, err := a.parseMetadata(meta)
+	if err != nil {
+		return errors.New("failed to parse metadata: " + err.Error())
+	}
+
+	err = a.Authenticate(ctx, m)
+	if err != nil {
+		return errors.New("failed to authenticate with Akeyless: " + err.Error())
+	}
+
+	return nil
+}
+
+// Authenticate authenticates with Akeyless using the provided metadata.
+// It returns an error if the authentication fails.
+func (a *akeylessSecretStore) Authenticate(ctx context.Context, metadata *akeylessMetadata) error {
+
+	a.logger.Debug("Creating authentication request to Akeyless...")
+	authRequest := akeyless.NewAuth()
+	authRequest.SetAccessId(metadata.AccessID)
+	authRequest.SetAccessType(metadata.AccessType)
```

Comment:
i think this is missing in metadata.yaml. Pls use the allowed values field too for this one for the consts you set for the select case below